scripts/install_requirements.sh

```diff
@@ -1,10 +0,0 @@
-#!/bin/bash
-
-pip install -r requirements.txt
```

Comment:
Don't remove this script please - it's useful when installing into a virtual environment (e.g not docker). The `spacy download en_core_web_sm` is still required and I think the `punkt` data package from NLTK is used in the Stemmer still too.

Reply:
Oh right, you don't want to have to download the spacy model. Personally I don't like this approach (e.g it makes what packages are actually required kind of opaque if we download them at runtime) but if others think it is fine then sure.

Reply:
I'm happy to leave the script and add `spacy download en_core_web_sm`--but I don't want to add `punkt` back unless I can identify where it's being used.  It's not being used in any of our test cases.  I wanted to remove both of those in this PR so I could run the tests in a clean environment.

Reply:
It looks like we "use it" in the unused and untested:

```
$ git grep NltkWordSplitter
allennlp/data/tokenizers/word_splitter.py:class NltkWordSplitter(WordSplitter):
```

Reply:
It's used here: https://github.com/allenai/allennlp/blob/master/allennlp/data/tokenizers/word_splitter.py#L141

If you want to remove this tokeniser and remove this data download that's fine with me.

Reply:
I will in a separate PR.